### PR TITLE
File upload ux

### PR DESCRIPTION
**Pull request by DarksightKellar**
This PR updates the large file PR to show the upload and download progress to and from arweave of larger files.

### CODE REVIEW

File: src/features/embalm/stepContent/hooks/useBundlr.ts [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
   const toast = useToast();
 
   // Pull some bundlr data from store
-  const { bundlr, txId, isConnected, isFunding } = useSelector(x => x.bundlrState);
+  const { bundlr, isFunding } = useSelector(x => x.bundlrState);
 
   // Used to tell the component when to render loading circle
   const [isWithdrawing, setIsWithdrawing] = useState(false);
 
+  const [chunkedUploader, setChunkedUploader] = useState<ChunkingUploader>();
```
Comment (DarksightKellar) [inline src/features/embalm/stepContent/hooks/useBundlr.ts:28]: Most of the updates on this file is meant to decouple uploader and event listener setup from the actual call to upload. Previous version was causing state conflicts and preventing the progress state (even in global state) from being read on the presentation layer.

File: src/features/embalm/stepContent/hooks/useCreateSarcophagus/useCreateSarcophagus.ts [hunk 2]
```diff
@@ -23,16 +23,33 @@ export function useCreateSarcophagus(
...
+    return new Map<CreateSarcophagusStage, string>([
+      [CreateSarcophagusStage.DIAL_ARCHAEOLOGISTS, ''],
+      [CreateSarcophagusStage.ARCHAEOLOGIST_NEGOTIATION, ''],
+      [CreateSarcophagusStage.UPLOAD_PAYLOAD, uploadStep],
+      [CreateSarcophagusStage.APPROVE, ''],
+      [CreateSarcophagusStage.SUBMIT_SARCOPHAGUS, 'Waiting for transaction'],
+      [CreateSarcophagusStage.CLEAR_STATE, ''],
+      [CreateSarcophagusStage.COMPLETED, ''],
+    ]);
+  }, [uploadStep]);
```
Comment (DarksightKellar) [inline src/features/embalm/stepContent/hooks/useCreateSarcophagus/useCreateSarcophagus.ts:47]: There's probably a better way of doing this, but this map is meant to be used to show extra UI information during a stage that takes a little longer than normal. Used similarly to the stage->stageFunction mapping

File: src/hooks/useArweave.ts [hunk 3]
```diff
@@ -81,11 +92,49 @@ export function useArweave() {
...
         console.log('error', error);
-
         throw new Error(`Error fetching arweave file: ${error}`);
       }
     },
-    [arweave]
+    [arweave, onDownloadProgress]
+  );
+
+  const customGetTx = useCallback(
```
Comment (DarksightKellar) [inline src/hooks/useArweave.ts:97]: This function contains the same code as `arweave.transactions.get(arweaveTxId)`, which is used on mainnet. Only diff is the addition of `onDownloadProgress` to the `api.get` config called on `108`.

File: src/features/embalm/stepContent/hooks/useBundlr.ts [hunk 4]
```diff
@@ -82,44 +89,109 @@ export function useBundlr() {
...
-        throw new Error('Bundlr not connected');
+  let rejectUploadPromise = useRef<any>();
+  let resolveUploadPromise = useRef<any>();
+
+  useEffect(() => {
+    if (!chunkedUploader || !fileBuffer || !rejectUploadPromise) return;
+
+    chunkedUploader.setChunkSize(5_000_000);
+
+    chunkedUploader?.on('chunkUpload', chunkInfo => {
```
Comment (sethhrbek) [inline src/features/embalm/stepContent/hooks/useBundlr.ts:100]: Can the `?` be removed here? There is a undefined check on line 96

File: src/features/embalm/stepContent/hooks/useBundlr.ts [hunk 5]
```diff
@@ -82,44 +89,109 @@ export function useBundlr() {
...
+      const chunkedUploadProgress = chunkInfo.totalUploaded / fileBuffer.length;
+      dispatch(setUploadProgress(chunkedUploadProgress));
+    });
+
+    chunkedUploader?.on('chunkError', e => {
+      const errorMsg = `Error uploading chunk number ${e.id} - ${e.res.statusText}`;
+      console.error(errorMsg);
+      rejectUploadPromise.current(errorMsg);
+      dispatch(setIsUploading(false));
+    });
```
Comment (sethhrbek) [inline src/features/embalm/stepContent/hooks/useBundlr.ts:110]: If there is an error in the UX, is a user able to click the "retry" button to restart the upload? And in which case, does that clear this error message?

**Review by sethhrbek — APPROVED** [review]

**Pull request merged**